src/models/flinkComputePool.ts

```diff
@@ -0,0 +1,87 @@
+import { ThemeIcon, TreeItem, TreeItemCollapsibleState } from "vscode";
+import { ConnectionType } from "../clients/sidecar";
+import { CCLOUD_CONNECTION_ID, IconNames } from "../constants";
+import { CustomMarkdownString } from "./main";
+import { ConnectionId, EnvironmentId, IResourceBase, isCCloud, ISearchable } from "./resource";
+
+export abstract class FlinkComputePool implements IResourceBase, ISearchable {
+  abstract connectionId: ConnectionId;
+  abstract connectionType: ConnectionType;
+  iconName: IconNames = IconNames.FLINK_COMPUTE_POOL;
+
+  environmentId!: EnvironmentId;
+
+  id!: string;
+  name!: string;
+
+  searchableText(): string {
+    return `${this.name} ${this.id}`;
+  }
+}
+
+export class CCloudFlinkComputePool extends FlinkComputePool {
+  readonly connectionId: ConnectionId = CCLOUD_CONNECTION_ID;
+  readonly connectionType: ConnectionType = ConnectionType.Ccloud;
+
+  provider!: string;
+  region!: string;
+  maxCfu!: number;
```

Comment:
May end up being a field we push to `FlinkComputePool` instead of being CCloud-specific